### DISCUSSION TITLE
graphscope复现ldbc性能测试

### DISCUSSION THREAD

**Discussion by timer-2117** [alibaba/GraphScope#3124] [Q&A] [answered]
在执行benchmark性能测试时有几个问题想问下
1、在warmup阶段执行13h，并且有很多延迟（查看机器的网络、磁盘性能均不亚于ldbc官方测试的机器配置）
<img width="863" alt="90879e1e115a9bb622c97a671d0d70b" src="https://github.com/alibaba/GraphScope/assets/53077051/d6854c50-b4e7-4796-b938-10ff124e97ea">
2、在排查原因时，看到在执行测试期间机器的cpu使用比较低
![image](https://github.com/alibaba/GraphScope/assets/53077051/47f78b74-037c-4a8c-a8e9-6f75575375d8)

**Comment by luoxiaojian** [comment, marked as answer]
这个我能想到的可能的原因有：
1. 启动rt_server时，没有指定并行度
2. 存放WAL的介质/文件系统限制了并行性，因为我们的实现里是需要并行写WAL至文件的
3. 软硬件环境与我们测试的区别太大
4. 是否配置了aio-max-nr
 "echo "fs.aio-max-nr = 1048576" » etcsysctl.conf && sysctl -p etcsysctl.conf"

只从你给的这些信息我也没法很明确地定位到问题，和给出解决办法
如果你觉得有必要，我可以给你一个能复现3w吞吐率的环境，你可以对照着看一下哪些配置不一致

**Reply by timer-2117** [reply]
很奇怪的一点是当我重新又执行的时候 这次的cpu也上去了 warmup的时间也算短到合理时间半小时了，实际上我什么都没有更改。测试环境对比了不会又很大区别。启动rt_server时 -s 48指定了并行度，因为二次运行时正常应该不是文件系统的问题吧 ；第四点我查看了这个文件仅有kernel.core_pattern=core
kernel.kptr_restrict=0两个配置，请问是这个的问题吗，但是为什么我二次运行时就正常了呢很疑惑

**Reply by timer-2117** [reply]
确认了一下，在rt_server所在机器上有fs.aio-max-nr = 1048576，在driver上没有

**Reply by luoxiaojian** [reply]
这个可能的原因比较多吧，如果是云上的服务器，它和磁盘的连接可能是通过网络，磁盘可能也是虚拟出来的

**Reply by luoxiaojian** [reply]
server侧配置了fs.aio-max-nr就可以

**Reply by timer-2117** [reply]
好的 谢啦 我再看看
